src/Composer/Installers/BaseInstaller.php

```diff
@@ -20,6 +25,11 @@ public function __construct(PackageInterface $package = null, Composer $composer
     {
         $this->composer = $composer;
         $this->package = $package;
+
+        if (!is_null($this->package))
+        {
+            $this->io = $this->composer->getInstallationManager()->getInstaller($this->package->getType())->getIO();
```

Comment:
You could pass the IO object in the constructor at https://github.com/composer/installers/blob/master/src/Composer/Installers/Installer.php#L90 and https://github.com/composer/installers/blob/master/src/Composer/Installers/Installer.php#L158 then you don't need to jump through these hoops.